src/main/java/org/opentripplanner/middleware/utils/HttpResponseHandler.java

```diff
@@ -0,0 +1,35 @@
+package org.opentripplanner.middleware.utils;
+
+import org.apache.http.HttpResponse;
+import org.apache.http.client.ResponseHandler;
+import org.apache.http.util.EntityUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+/**
+ * Extracts the required values from an {@link HttpResponse} and populate an instance of {@link HttpResponseValues}.
+ */
+public class HttpResponseHandler implements ResponseHandler<HttpResponseValues> {
+    private static final Logger LOG = LoggerFactory.getLogger(HttpResponseHandler.class);
+
+    public HttpResponseValues handleResponse(final HttpResponse response) {
+        return new HttpResponseValues(response, getResponseBodyAsString(response));
+    }
+
+    /**
+     * Extract the response body as a String. If an exception occurs return null.
+     */
+    private static String getResponseBodyAsString(HttpResponse response) {
+        String responseBody = null;
+        try {
+            responseBody = EntityUtils.toString(response.getEntity());
+            EntityUtils.consume(response.getEntity());
+        } catch (IOException e) {
+            LOG.error("Unable to get response body", e);
+        }
+        return responseBody;
+    }
```

Comment:
Is it required to parse the response body and consume it here, or can that be done as an instance method in `HttpResponseValues`? The reason I ask is that sometimes with large response bodies, it may be more efficient to read the response body as a stream instead of first reading all of it into a string and then doing further processing.

Reply:
Unfortunately not, the connection is closed after this point so any attempts to down the response body would fail.